design/baremetal-operator/bulk-set-bios-config.md

```diff
@@ -0,0 +1,228 @@
+up a machine using that method and then push the configuration to multiple
+machines as described here.
+
+Most hardware vendors support a way to apply bulk settings, often by manually
+configuring one host, exporting those settings in a vendor-specific file
+format, and then importing that file on the BMCs of other hosts. Accepting that
+the format for these profiles is vendor-specific eliminates a lot of the
+implementation complexity, at the risk of moving that complexity to the
+end-user. This requirement is similar to something Dell has proposed in the
+Ironic community
+([hw config](https://review.opendev.org/c/openstack/ironic-specs/+/740721))
+to export and import an entire system iDRAC inventory including RAID and
+networking requirements. Compared to that approach, this proposal attempts
+a more lightweight, vendor-neutral approach just for BIOS configuration.
+
+### Goals
+
+Provide a vendor-neutral approach of updating BIOS configuration in a bulk
+manner across all similar hardware. Create a template for the current
+BIOS configurations settings cached on a server. Give access to the BIOS
+registry which can be used as documentation so a user can easily update
+the BIOS settings to desired values. Validate the BIOS Configuration
+values based on the vendor's BIOS Registry before sending to the BMC.
+
+### Non-Goals
+
+This proposal does not attempt to fulfill the following goals:
+
+- Allow the transfer of BIOS configuration between different vendor’s
+  servers or different vendor models
+- Support bulk set of other configurations besides BIOS
+- Provide storage mechanism for BIOS settings
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+When provisioning a large amount of machines with identical settings, our
+customers want to provide a known, validated payload with all the desired
+firmware configuration.
+
+This ensures all the machines will have the identical settings, reducing issues
+caused by unexpected configuration differences.
+
+Since the name/value pair mapping is vendor specific and not intuitive, it
+doesn’t lend itself to creating the input by hand using existing vendor
+documentation, except for potentially a small subset of names. It’s desirable
+to take an existing BIOS config set and modify the current settings to the
+desired settings. The BIOS Registry can be used as a source of documentation
+to modify the settings.
+
+## Design Details
+
+This proposes a new Custom Resource Definition (CRD) to store the settings
+read from Ironic along with the modified settings to write to Ironic. Since
+each vendor provides 100 or more BIOS Settings, its not practical to store
+these in the BareMetalHost CRD. The new CRD will be named
+`HostFirmwareSettings` and will consist of the following:
+
+- `currentFirmwareSettings` - the current BIOS Settings retrieved from Ironic
+  via the bios API will be stored in the Status section. This dataset includes
```

Comment:
A link to the `bios` API here could be useful

Reply:
Added.